### PR TITLE
feat(vital-carousel): CSS Carousel with minimaljs

### PR DESCRIPTION
**Pull request by hvanyo**
## Changes

- Add carousel component implemented as mostly css and minimaljs

## Test Instructions

- styleguide/carousel
- https://bodiless-js-next-1mflmmy5v-jnj-demo-account.vercel.app/styleguide/carousel/

### CODE REVIEW

File: packages/vital-carousel/src/components/Carousel/VitalCarouselClean.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+type VitalCarouselBaseProps = DesignableComponentsProps<VitalCarouselComponents>;
+
+const carouselActive = '#019881'; // vitalCarouselTokens.ScrollIndicatorInactive
+const carouselInActive = '#BFBFBF'; // vitalCarouselTokens.ScrollIndicatorActive;
+
+const cssString = `:root { --carouselActive: ${carouselActive}; --carouselInActive: ${carouselInActive}; }`;
+
+const CarouselScript = () => (
+  <Helmet>
+    <script type="text/javascript">
```
Comment (hvanyo) [inline packages/vital-carousel/src/components/Carousel/VitalCarouselClean.tsx:19]: I need to set active/inactive colors in my tailwind file so I set up css variables.   I thought I had it figured out cause this worked on npm run dev but not on static.   If on static, I manually add this to head on static <style>:root { --carouselActive: #019881; --carouselInActive: #BFBFBF; }</style> then it works fine. ⁠

Comment (marcopagliarulo) [reply packages/vital-carousel/src/components/Carousel/VitalCarouselClean.tsx:19]: I think the best way is to define the classes on tailwind config using @apply to apply tailwind classes to a custom class https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply

File: packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+import type { VitalCarousel } from '../types';
+import { CAROUSEL_NODE_KEY } from '../../utils/constants';
+import CarouselSlide from '../../utils/CarouselSlide';
+import { CarouselThumbClean, vitalCarouselThumb } from '../../CarouselThumb';
+import { CarouselDotClean, vitalCarouselDot } from '../../CarouselDot';
+
+// Using withDesign throughout file to target the list that is added by asBodilessList.
+
+const Default = asVitalCarouselToken({
+  // Core: {
```
Comment (hvanyo) [inline packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts:21]: @marcopagliarulo  should it go on like these or elsewhere?

File: packages/vital-carousel/src/components/Carousel/VitalCarouselClean.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+const sliderSimpleInit = isUndefined; // Seems needed or get errors on line 35/36
+
+const withCarouselInit: HOC = Component => {
+  const WithCarouselInit = (props: any) => {
+    const carouselRef = useRef<HTMLElement>(null);
+
+    useLayoutEffect(() => {
+      // if (sliderSimpleInit !== 'undefined') {
+      sliderSimpleInit(carouselRef.current);
```
Comment (hvanyo) [inline packages/vital-carousel/src/components/Carousel/VitalCarouselClean.tsx:36]: @marcopagliarulo sliderSimpleIniti was saying not intitialized..
So I set to isUndefined and fixed the error on 36 but still had an error on 35..

Comment (marcopagliarulo) [reply packages/vital-carousel/src/components/Carousel/VitalCarouselClean.tsx:36]: if you are referring to the typescript error, you should solve it adding 
`declare const sliderSimpleInit: function;`
to https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/%40types/globals.d.ts

File: packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts (hunk 2 above)
Comment (marcopagliarulo) [reply packages/vital-carousel/src/components/Carousel/tokens/vitalCarousel.ts:21]: Applied to the Wrapper should be fine, logically it should go under domain Behaviour

**Comment by salikn** [comment]
@hvanyo  this looks good to me, the only thing i see is when you are on desktop browser and resize the window to smaller width then try to use the mouse to scroll through the slides. This does not work, It works only when using toggle device (seems it only attaches to touch/drag events not mouse)

**Review by salikn — APPROVED** [review]

**Review by KaterinaPashkevich — APPROVED** [review]

**Review by VancheeZze — DISMISSED** [review]

**Review by samlynndavis — APPROVED** [review]

**Pull request merged**